### DISCUSSION TITLE
How to ensure that code follows official style guide?

### DISCUSSION THREAD

**Discussion by bitbrain** [bitbrain/beehave#136] [Ideas]
Currently, code style seems to be inconsistent and some contributors already have made a great effort to use the [official code style guide](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html). A few ideas:

1. it would still be useful to have some sort of step in any PR that validates that PRs follow that notion. A problem with this is that folks may notice too late so the "feedback loop" would be too long. See Point 2. how we could solve that
2. Perhaps, there could be a pre-commit hook of some sort. It might become annoying for people, so offering a way to "auto-format" could also be an option.
3. Maybe there is a way to introduce a linter inside Godot Engine itself that auto-formats on save?

**Comment by Wichamir** [comment]
@bitbrain [This](https://github.com/Scony/godot-gdscript-toolkit) might be useful.
